channels/routing.py

```diff
@@ -158,3 +235,24 @@ async def __call__(self, scope, receive, send):
             raise ValueError(
                 "No application configured for channel name %r" % scope["channel"]
             )
+
+
+def reverse(*args, urlconf=None, **kwargs):
+    """reverse wrapper for django's reverse function
+
+    Parameters
+    ----------
+    urlconf : str, optional
+        The root path of the routings, by default None
+
+    See the django's [reverse](https://docs.djangoproject.com/en/5.0/ref/urlresolvers/#reverse)
+    for more details of the other arguments
+
+    Returns
+    -------
+    str
+        The reversed url
+    """
+    if urlconf is None:
+        urlconf = settings.ROOT_WEBSOCKET_URLCONF
+    return django_reverse(*args, urlconf=urlconf, **kwargs)
```

Comment:
I think this is a change that deserves its own discussion. I don't think we should introduce a new  `ROOT_WEBSOCKET_URLCONF` setting. Please revert the addition of this function.

Reply:
I want to discuss this. (Sorry for long reply...)
When you don't set the `ROOT_WEBSOCKET_URLCONF`, this `reverse` function will be limited. Users must specify the `urlconf` as the parent `routings.py` every time.

- Why every time?

`settings.ROOT_URLCONF` should be configured in the original django's routing system. And actually, django's `reverse` function refers to `ROOT_URLCONF` here

https://github.com/django/django/blob/f16a9a556fb4225f9094048614f4fcec3db7e067/django/urls/base.py#L30

```python
def resolve(path, urlconf=None):
    if urlconf is None:
        urlconf = get_urlconf()
    return get_resolver(urlconf).resolve(path)


def reverse(viewname, urlconf=None, args=None, kwargs=None, current_app=None):
    if urlconf is None:
        urlconf = get_urlconf()
    resolver = get_resolver(urlconf)
```

This means original django's routing system parses all configured urlpatterns from `ROOT_URLCONF`. That's why we can use `reverse` function.

Example;

```txt
src
├── app1
│   ├── urls.py
│   └── routings.py
├── urls.py (root)
└── routings.py (root)
```

In `settings.py`

```python
ROOT_URLCONF = 'src.urls' # this is entrypoint to parse all urls
```

That's why if we want to use `reverse` for channels, we should configure the "entrypoint" to parse all urls of channels.

Reply:
I totally see where you're coming from, but I don't want to mix in those new changes into this PR. I see this as 2 separate issues:

1. Support nested urlpatterns
2. Enable `reverse`

This PR is (1). (2) requires more discussion and design, so I'd like to make that a separate PR. Do you mind splitting them up so we can discuss (2) in greater depth?

Reply:
OK!